operator/pkg/manifest/helm_client_cache.go

```diff
@@ -36,10 +39,29 @@ func (h *HelmClientCache) Get(key client.ObjectKey) types.HelmClient {
 
 // Set saves the passed HelmClient into HelmClientCache for the client.ObjectKey.
 func (h *HelmClientCache) Set(key client.ObjectKey, helmClient types.HelmClient) {
-	h.cache.Store(key, helmClient)
+	h.clientCache.Store(key, helmClient)
 }
 
 // Delete deletes the HelmClient from HelmClientCache for the passed client.ObjectKey.
 func (h *HelmClientCache) Delete(key client.ObjectKey) {
-	h.cache.Delete(key)
+	h.clientCache.Delete(key)
+}
+
+// GetMemCachedClient loads the CachedDiscoveryInterface from HelmClientCache for the passed client.ObjectKey.
+func (h *HelmClientCache) GetMemCachedClient(key client.ObjectKey) discovery.CachedDiscoveryInterface {
```

Comment:
To have clean cache separation you should probably introduce 2 separate caches so that you cannot mixup a key collision